src/mach.rs

```diff
@@ -826,7 +865,11 @@ impl<'a> Mach<'a> {
         // write cstrings
         //////////////////////////////
         for cstring in self.cstrings {
-            file.write_all(cstring.data)?;
+            if let Data::Blob(bytes) = cstring.data {
+                file.write_all(bytes)?;
+            } else {
+                unreachable!();
```

Comment:
The problem I keep running into is that most of these types are set up during the _declaration_ of the data, not the _definition_. To encode this in the type system, we would have to create a new `cstring` struct during definition, which since we have to do it for a bunch of different types would make everything harder to understand IMO.

Maybe I'm just not understanding the current code very well.

Reply:
Although actually now that I look at it you're right, it's definitely possible to have a `ZeroInit` string: https://github.com/m4b/faerie/pull/99/files#diff-0a6142b10f08148adf63ae919efb879fR700

Reply:
Fixed the logic bug but it still has the `unreachable`.